packages/components/src/components/bal-tabs/components/steps.tsx

```diff
@@ -80,7 +82,7 @@ export const StepList: FunctionalComponent<TabProps> = ({ value, float, clickabl
                 }}
                 style={{ display: !tab.done ? 'block' : 'none' }}
               >
-                {tab.failed ? '!' : index + 1}
+                {tab.failed ? '!' : tab.hidden ? index : index++}
```

Comment:
if the step is hidden then don't increase the index, but for the other status increase it 

- normal
- failed
- done